client/components/external-link/index.jsx

```diff
@@ -36,7 +36,7 @@ export default React.createClass( {
 			'has-icon': !! this.props.icon,
 		} );
 
-		const props = assign( {}, this.props, {
+		const props = assign( {}, omit( this.props, 'icon', 'iconSize' ), {
```

Comment:
Yes, I am in favor of whitelisting as well.
The added benefit is that its much clearer to trace where the props came from